easybuild/easyblocks/h/hdf5.py

```diff
@@ -38,9 +38,19 @@
 import easybuild.tools.toolchain as toolchain
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
 from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import write_file
 from easybuild.tools.modules import get_software_root
 from easybuild.tools.systemtools import get_shared_lib_ext
 
+# pkgconfig data template
+HDF5_PKG_CONFIG = """Name: HDF5
+Description: Hierarchical Data Format 5 (HDF5)
+Version: %s
+Requires:
+Cflags: -I%s
+Libs: -L%s -lhdf5
```

Comment:
@akesandgren Did you come up with this yourself, or is there an upstream reference for it somewhere?

It looks OK, but I do wonder if this will cause problems somewhere because we have a `.pc` in place all of a sudden...

What about dependencies like `Szip` etc., should we take those into account here too?

Reply:
Based on Debian packaging file(s). (Added to the comment)
Yes, it should most likely contain linking with zlib and Szip, but also libs from any other dependency a specific easyconfig uses, like libxml2.

Didn't want to make it too complicated to start with though.

Reply:
I understand, but won't this possible impact stuff that depends on `HDF5`?

I guess testing will tell...

Reply:
It should be done properly before merging, but I haven't gotten around to test it fully yet.
Don't really know what to test it with to see if -lz -lszip is needed.

Reply:
The current definition of only including `-lhdf5` worked fine with `h5py` (but I'm not sure the `.pc` is even picked up there).

Making this more complete shouldn't make things worse?

We should dynamically complete this though, something:

```python
known_deps = {
    'libxml2': '-lxml2',
    'Szip': '-lsz',  # not sure what's correct here
    'zlib': '-lz',
}
for dep in known_deps:
    # compose value for 'Libs' using known_deps[dep]
```

or even:

```python
for dep in self.cfg.dependencies:
  if dep in known_deps:
    # compose value for 'Libs' using known_deps[dep]
  else:
    raise EasyBuildError("Unknown dependency for HDF5, don't know how to update pkgconfig file: %s", dep)